app/admin/users.rb

```diff
@@ -0,0 +1,19 @@
+ActiveAdmin.register User do
+
+  # See permitted parameters documentation:
+  # https://github.com/activeadmin/activeadmin/blob/master/docs/2-resource-customization.md#setting-up-strong-parameters
+  #
+  # Uncomment all parameters which should be permitted for assignment
+  #
+  # permit_params :first_name, :last_name, :email, :password_digest, :confirmed_at, :unconfirmed_email, :role
+  #
+  # or
+  #
+  permit_params do
+    permitted = [:first_name, :last_name, :email, :password_digest, :confirmed_at, :unconfirmed_email, :role]
+    permitted << :other if params[:action] == 'create' && current_user.admin?
```

Comment:
You do not need to check like this, only `AdminUser` can access admin panel. And also current_user will also not relevant here, because for `admin_user` we have `current_admin_user` method.

Reply:
okay sir